core/variant/variant.h

```diff
@@ -715,6 +715,10 @@ class Variant {
 	static bool has_constant(Variant::Type p_type, const StringName &p_value);
 	static Variant get_constant_value(Variant::Type p_type, const StringName &p_value, bool *r_valid = nullptr);
 
+	static void get_enums_for_type(Variant::Type p_type, List<StringName> *p_enums);
```

Comment:
So I have a doubt here, is this not redundant with constants? Would it not be better to simply have a method like `get_enum_constants` besides `get_enums_for_type` ?

Reply:
I don't think so. There is a hierarchy level in more with enum.  
It is kind of namespaced constants, but I still need to specify for which "namespace" I want to list.